Add USAGE.md documentation (#64)
sclorg/testing-farm-as-github-action: USAGE.md

# How to use Testing Farm as GitHub Action in your projects

Before start of using Testing Farm as GitHub Action in your projects,
there are several steps which you have to do it.

## As Testing Farm team for public or private ranch

> Before any implementation as Testing Farm team for API_KEY here [onboarding](https://docs.testing-farm.io/general/0.1/onboarding.html).

Reviewer: s/As/Ask/ in the headline and also in the paragraph. It might be good to explain the difference between public and private ranch here, when it's used here.

Author: Good point.

Author: Fixed by https://github.com/sclorg/testing-farm-as-github-action/pull/64/commits/cf1e8e968831ae6d1a95284a5838c2c152d46185 and https://github.com/sclorg/testing-farm-as-github-action/pull/64/commits/d3ab80538320df2b1fb56b74361edad681d00dd7

Reviewer 2: This does not seem to be fixed yet.

Author: See here: https://github.com/sclorg/testing-farm-as-github-action/pull/64/commits/d3ab80538320df2b1fb56b74361edad681d00dd7 at the beginning.

Author: I have rebased the pull request.

Author: https://github.com/sclorg/testing-farm-as-github-action/pull/64/files#diff-58a825c70bccfbe691d7ddbddb133275f4f657de6aeb4a502fa9fce279f8bf8dR9

Reviewer 2: I do not see it here.

## Setup TMT plan in GitHub / GitLab repos

In your GitHub or GitLab repos create:
* [.fmf](https://github.com/sclorg/sclorg-testing-farm/tree/main/.fmf) directory with file [version](https://github.com/sclorg/sclorg-testing-farm/blob/main/.fmf/version) which contains 1.
* [plans](https://github.com/sclorg/sclorg-testing-farm/tree/main/plans) directory which will contain TMT plans for Testing Farm
* create a TMT plan like it is already done here [c9s](https://github.com/sclorg/sclorg-testing-farm/blob/main/plans/c9s.fmf)

> * The TMT plan contains instructions how to setup, install and execute tests on quest provided by Testing Farm. For more information see [documentation](https://tmt.readthedocs.io/en/stable/examples.html#plans) or take an inspiration from our [plans](https://github.com/sclorg/sclorg-testing-farm/tree/main/plans) diretory

Reviewer: Would it be possible to provide an example of how to run the tmt locally for testing the plan? It is in the `tmt` documentation, but a quick-start one-liner might be helpful..

Author: Fixed by https://github.com/sclorg/testing-farm-as-github-action/pull/64/commits/d3ab80538320df2b1fb56b74361edad681d00dd7

## Setup GitHub Action

As soon as you have your TMT plan ready then let's create an GitHub Actions for your use case

### Testing by GitHub Actions

Create a directory `.github/workflows` in your GitHub repository.

> ### Testing in pull request

Reviewer: The wording "Testing in pull request" didn't make sense  to me until I read the description. So, it's technically correct, but maybe we can rewrite it to something more clear, like "Configuration for tests to be run on pull-requests"... or something similar.

Author: The new caption is `### Testing pull request changes via Testing Farm`

Create a file like `testing-farm-pr.yml`
The file can look like as is mentioned in [README.md](https://github.com/sclorg/testing-farm-as-github-action#example)

> ### Testing when pull request is merged

Reviewer: This one is clear, but "Testing" may lead to perception that only testing can be done, but we also can do some other production actions, like we do for sclorg containers. Maybe we can call this step as "Testing or other actions performed after a pull request is merged"?

Author: Fixed

# How to use Testing Farm as GitHub Action in your projects

Before start of using Testing Farm as GitHub Action in your projects,

> there are several steps which you have to do it.

Reviewer 2: ```suggestion
there are several steps that you have to do.
```

## Ask Testing Farm team for public or private ranch

Before any implementation as Testing Farm team for API_KEY here [onboarding](https://docs.testing-farm.io/general/0.1/onboarding.html).
Public ranch is used for testing your upstream repository on Fedora, CentOS 7, and CentOS Stream guests.

Private ranch is used for testing your upstream repository on RHEL and Oracle-linux guests.

## Setup TMT plan in GitHub / GitLab repos

> In your GitHub or GitLab repos create:

Author: I will add this note after the list. Good point.
